### PR TITLE
Introduce `label` in expose

### PR DESCRIPTION
**Pull request by Drafteed**
Based on https://github.com/Koenkk/zigbee2mqtt/issues/18445#issuecomment-1662899667 and https://github.com/Koenkk/zigbee2mqtt/pull/16702#issuecomment-1438906709

Docs PR: https://github.com/Koenkk/zigbee2mqtt.io/pull/2170

This part does not contain manual override labels for devices and presets.

### CODE REVIEW

**Comment by Koenkk** [comment]
Looks good!

Please also make a PR for the docs repo: https://github.com/Koenkk/zigbee2mqtt.io/blob/master/docs/guide/usage/exposes.md

**Comment by Koenkk** [comment]
Note to self; after this merge https://github.com/Koenkk/zigbee2mqtt.io/pull/2170

File: src/lib/exposes.ts [hunk 1]
```diff
@@ -12,6 +13,7 @@ export class Base {
     type: 'switch' | 'lock' | 'binary' | 'list' | 'numeric' | 'enum' | 'text' | 'composite' | 'light' | 'cover' | 'fan' | 'climate';
     endpoint?: string;
     property?: string;
+    label?: string;
```
[the remark this answers is not in the document]
Comment (Drafteed) [reply src/lib/exposes.ts]: Done.

**Comment by Koenkk** [comment]
Thanks! Since this is merged now, could you also update `homeassistant.ts` to use it here?

**Comment by Drafteed** [comment]
> Thanks! Since this is merged now, could you also update `homeassistant.ts` to use it here?

Sure, it's almost ready.

**Pull request merged**